graphql_compiler/schema_transformation/utils.py

```diff
@@ -157,27 +202,93 @@ class NoOpRenamingError(SchemaTransformError):
       the schema named type_name
     * type_renamings is iterable and maps a string type_name to itself, i.e.
       type_renamings[type_name] == type_name
+    * There exists an object type T named type_name in the schema such that
+      field_renamings[type_name] is both iterable and contains a string field_name but there doesn't
+      exist a field named field_name belonging to T in the schema.
+    * field_renamings is iterable and contains a string [type_name] but there doesn't exist an
+      object type in the schema named type_name
+    * There exists an object type T named type_name in the schema such that
+      field_renamings[type_name] is both iterable and 1:1 maps a string field_name to itself within
+      a particular type, i.e. field_renamings[type_name][field_name] == [field_name]
     """
 
     no_op_type_renames: Set[str]
+    no_op_nonexistent_type_field_renames: Set[str]
+    no_op_field_renames: Dict[str, Set[str]]
 
-    def __init__(self, no_op_type_renames: Set[str]) -> None:
+    def __init__(
+        self,
+        no_op_type_renames: Set[str],
+        no_op_field_renames: Dict[str, Set[str]],
+        no_op_nonexistent_type_field_renames: Set[str],
+    ) -> None:
         """Record all no-op renamings."""
-        if not no_op_type_renames:
+        # TODO(Leon): Would like feedback on this-- I notice a pattern where I create error message
+        # objects that must have at least one of these parameters as non-empty. Is there a better
+        # way to ensure this than manually writing this verification (which needs to change as the
+        # parameters change)? This also comes up when deciding whether or not to raise these errors
+        # as well-- for instance, checking if any of a number of error-collecting data structures
+        # are nonempty, and raising an error if any are.
```

Comment:
only makes it slightly better, but maybe using `not any([...])` where `[...]` is the list of things to check?